File: homework01/caesar.py
def encrypt_caesar(plaintext: str, shift: int = 3) -> str:
    """
    Encrypts plaintext using a Caesar cipher.

    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    ciphertext = ''

    for character in plaintext:
        if character.islower():
            chr_code = ord('a')
            ciphertext += chr(chr_code + (ord(character) - chr_code + (shift % 26)) % 26)
        elif character.isupper():
            chr_code = ord('A')
            ciphertext += chr(chr_code + (ord(character) - chr_code + (shift % 26)) % 26)
        else:
            ciphertext += character

    return ciphertext


def decrypt_caesar(ciphertext: str, shift: int = 3) -> str:
    """
    Decrypts a ciphertext using a Caesar cipher.

    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    plaintext = ''

    for character in ciphertext:
        if character.islower():
            chr_code = ord('a')
            plaintext += chr(chr_code + (ord(character) - chr_code - (shift % 26)) % 26)
        elif character.isupper():
            chr_code = ord('A')
            plaintext += chr(chr_code + (ord(character) - chr_code - (shift % 26)) % 26)
        else:
            plaintext += character

    return plaintext

File: homework01-go/caesar/caesar.go
package caesar

import (
	"unicode"
)

func EncryptCaesar(plaintext string, shift int) string {
	var ciphertext string

	// Default value
	if shift == 0 {
		shift = 3
	}

	for _, character := range plaintext {
		if unicode.IsLower(character) {
			chrCode := int('a')
			ciphertext += string(rune(chrCode + (int(character)-chrCode+(shift%26))%26))
		} else if unicode.IsUpper(character) {
			chrCode := int('A')
			ciphertext += string(rune(chrCode + (int(character)-chrCode+(shift%26))%26))
		} else {
			ciphertext += string(character)
		}
	}

	return ciphertext
}

func DecryptCaesar(ciphertext string, shift int) string {
	var plaintext string

	// Default value
	if shift == 0 {
		shift = 3
	}

	for _, character := range ciphertext {
		if unicode.IsLower(character) {
			chrCode := int('a')
			newLetter := (int(character) - chrCode - (shift % 26)) % 26
			if newLetter < 0 {
				newLetter += 26
			}
			plaintext += string(rune(chrCode + newLetter))
		} else if unicode.IsUpper(character) {
			chrCode := int('A')
			newLetter := (int(character) - chrCode - (shift % 26)) % 26
			if newLetter < 0 {
				newLetter += 26
			}
			plaintext += string(rune(chrCode + newLetter))
		} else {
			plaintext += string(character)
		}
	}

	return plaintext
}
